### ISSUE TITLE
Transient network errors not retried

### ISSUE THREAD

**Issue by vieira** [googleapis/nodejs-common#473]
Although nodejs-common impements a retry mechanism using `retry-request`, the `shouldRetryRequest` method does not take into account transient network errors, example:

```
FetchError: request to ... failed, reason: getaddrinfo EAI_AGAIN pubsub.googleapis.com pubsub.googleapis.com:443
    at ClientRequest.<anonymous> (/app/index.js:104521:11)
    at ClientRequest.emit (events.js:198:13)
    at TLSSocket.socketErrorListener (_http_client.js:392:9)
    at TLSSocket.emit (events.js:198:13)
    at emitErrorNT (internal/streams/destroy.js:91:8)
    at emitErrorAndCloseNT (internal/streams/destroy.js:59:3)
    at process._tickCallback (internal/process/next_tick.js:63:19)
```
 https://github.com/googleapis/nodejs-common/blob/bb76f072c5320c30b8f6aa8b1b0cf514cc8e6a9d/src/util.ts#L514-L531

#### Environment details

  - OS: Linux (cos) on GKE; macOS 10.14.6
  - Node.js version: 10.16.3
  - yarn version: 1.17.3
  - `@google-cloud/common` version: 2.1.2

#### Steps to reproduce

  1. Make a request while network connection is down
  2. Notice that it immediately fails without any retry being made

Thanks!

**Comment by stephenplusplus** [comment]
Interesting. We do exclude those types of errors, but we must not be processing them correctly: https://github.com/stephenplusplus/retry-request/blob/master/index.js#L174

**Comment by stephenplusplus** [comment]
Looks like by default, retry-request makes a total of 3 attempts to a non-responsive endpoint, using a back-off delay. I'm not yet sure why the network errors aren't being retried like it is designed to.

**Comment by vieira** [comment]
@stephenplusplus Would it also honor the maxRetries option?

**Comment by stephenplusplus** [comment]
No, the preflight errors have a different counter.

**Comment by JustinBeckwith** [comment]
@stephenplusplus seems like you have the most context here.  Could I trouble you to take a look?

**Comment by bcoe** [comment]
@vieira I have been unable to recreate this issue, when I run the library with the network disconnected, I do see an error:

```bash
(node:40168) UnhandledPromiseRejectionWarning: FetchError: request to https://bigquery.googleapis.com/bigquery/v2/projects/long-door-651/datasets failed, reason: getaddrinfo ENOTFOUND bigquery.googleapis.com
```

But only after a connection has been retried three times, is this still a problem you're facing? If so would you be able to provide a minimal reproduction.

**Comment by vieira** [comment]
@bcoe Thank you for the fix in #534. Will this respect the `maxRetries` option in the `Service` constructor?
